### DISCUSSION TITLE
Github action for coding style

### DISCUSSION THREAD

**Discussion by Mohammadreza-73** [vitodeploy/vito#231] [Q&A]
Hi,
I have a question:
Why are you use Pint (PHP code style fixer) with `--test` option in [this file](https://github.com/vitodeploy/vito/blob/3b42f93654babfd1919726663945a249a6e2bdf4/.github/workflows/code-style.yml#L41) ?

Based on [Laravel documentation](https://laravel.com/docs/10.x/pint#running-pint) this option only inspect code for style error without actually changing the files.

You can fix your code style errors with Pint and then commit the changes with [auto commit](https://github.com/stefanzweifel/git-auto-commit-action) Github action, or use [StyleCI](https://styleci.io/) to Automatically fix the style of your PHP, JS and CSS code when pushing your code to origin.

**Comment by saeedvaziry** [comment]
> Why are you use Pint (PHP code style fixer) with --test option in [this file](https://github.com/vitodeploy/vito/blob/3b42f93654babfd1919726663945a249a6e2bdf4/.github/workflows/code-style.yml#L41) ?

It's there to make sure PRs don't get merged if they have incorrect coding style.

> use [StyleCI](https://styleci.io/) to Automatically fix the style

I am not sure StyleCI supports pint or not but generally not a fan of it

> commit the changes with [auto commit](https://github.com/stefanzweifel/git-auto-commit-action)

PRs are welcome to fix the code in the PRs' feature branches.
